colliderbox.cpp

```diff
@@ -0,0 +1,65 @@
+#include "colliderbox.h"
+
+#include <glm/glm.hpp>
+
+#include <iostream>
+#include <sstream>
+
+namespace pong
+{
+
+ColliderBox::ColliderBox(float width, float height)
+{
+    mBounds = {
+        glm::vec3(-width / 2.0f, -height / 2.0f, 0.0f),
+        glm::vec3( width / 2.0f,  height / 2.0f, 0.0f),
+        glm::vec3(-width / 2.0f,  height / 2.0f, 0.0f),
+        glm::vec3( width / 2.0f, -height / 2.0f, 0.0f)
+    };
+
+    mPositionBounds = mBounds;
+}
+
+void ColliderBox::UpdatePositionBounds(const glm::vec3& position)
```

Comment:
I am curious if there is a better way around this. This function should be called whenever the gameobject it is attached to updates its own position, so it will update here. However to call it from the gameobject it is public and be called anywhere. But realistically this should never be called from anywhere else. Is there is a good way to solve that?

Reply:
In terms of preventing it being called from anywhere else, I don't think there is a decent way to do it.

The technical way to do it would be making this private, then making the two classes `friend`s, but I think that is more evil than good.